straxen/corrections_services.py

```diff
@@ -130,14 +129,22 @@ def _get_correction(self, run_id, correction, version):
                 pmts = list(gains.keys())
                 for it_correction in pmts: # loop over all PMTs
                     if correction in it_correction:
-                        df = self.interface.read(it_correction)
+                        df = self.interface.read_by_index(it_correction, when)
+                        if self.check_for_nans(when, df, version):
+                            raise CMTnanValueError(f"For {it_correction} there are NaN values, this means no correction available "
+                                                   f"for {run_id} in version {version}, please check e-logbook for more info ")
+ 
                         if version in 'ONLINE':
                             df = self.interface.interpolate(df, when, how='fill')
                         else:
                             df = self.interface.interpolate(df, when)
                         values.append(df.loc[df.index == when, version].values[0])
             else:
-                df = self.interface.read(correction)
+                df = self.interface.read_by_index(correction, when)
+                if self.check_for_nans(when, df, version):
```

Comment:
if you use the read_by_index function the returned dataframe has only the two relevant items to this time so a simple df[version].isnull().values.any() makes more sense (if either of the two enclosing points are nan, raise error)

Reply:
yes, that make senses. thanks